packages/markdown-support/src/codeblock.ts

```diff
@@ -1,43 +1,44 @@
 import { visit } from 'unist-util-visit';
+import type { Node } from 'unist';
+import type { Element } from 'hast';
+import type { Code } from 'mdast';
 
 /**  */
 export function remarkCodeBlock() {
-  const visitor = (node: any) => {
-    const { data, meta } = node;
-    let lang = node.lang || 'html'; // default to html matches GFM behavior.
+  return function (tree: Node) {
+    visit(tree, 'code', (node: Code) => {
+      const { data, meta } = node;
+      let lang = node.lang || 'html'; // default to html matches GFM behavior.
 
-    let currentClassName = data?.hProperties?.class ?? '';
-    node.data = node.data || {};
-    node.data.hProperties = node.data.hProperties || {};
-    node.data.hProperties = { ...node.data.hProperties, class: `language-${lang} ${currentClassName}`.trim(), lang, meta };
-
-    return node;
+      let currentClassName = (data?.hProperties as { [k: string]: string })?.class ?? '';
+      node.data = node.data || {};
+      node.data.hProperties = node.data.hProperties || {};
+      node.data.hProperties = { ...(node.data.hProperties as { [k: string]: string }), class: `language-${lang} ${currentClassName}`.trim(), lang, meta };
+    });
   };
-  return () => (tree: any) => visit(tree, 'code', visitor);
 }
 
 /**  */
 export function rehypeCodeBlock() {
-  const escapeCode = (code: any) => {
-    code.children = code.children.map((child: any) => {
-      if (child.type === 'text') {
-        return { ...child, value: child.value.replace(/\{/g, 'ASTRO_ESCAPED_LEFT_CURLY_BRACKET\0') };
+  return function (tree: Node) {
```

Comment:
Using `Root` from `@types/hast` here should help narrow most of the types down!

Reply:
Done in 3392ebb